doc/model.rst

```diff
@@ -0,0 +1,683 @@
+2. `model.err` : will hold any errors that occurred during Model execution
+
+-----------------
+A Colocated Model
+-----------------
+A colocated model is when both the application and the
+database share compute resource. To create a colocated model,
+users first initialize a Model instance with the ``Experiment.create_model()`` function.
+A user must then colocate the model with the Model API function ``Model.colocate_db()``.
+This instructs SmartSim to launch an Orchestrator on the same compute
+nodes as the model when the model instance is deployed via ``Experiment.start()``.
+
+There are **three** different Model API helper functions to colocate a Model:
+
+- ``Model.colocate_db()``: An alias for `Model.colocate_db_tcp()`.
+- ``Model.colocate_db_tcp()``: Colocate an Orchestrator instance with this Model over TCP/IP.
+- ``Model.colocate_db_uds()``: Colocate an Orchestrator instance with this Model over UDS.
+
+Each function will initialize an unsharded database to the Model instance. Only the partnered Model
+may communicate with the database via a SmartRedis client by using the loopback TCP interface or
+Unix Domain sockets. Extra parameters for the database can be passed into the helper functions
+via kwargs.
+
+.. code-block:: python
+
+    example_kwargs = {
+        "maxclients": 100000,
+        "threads_per_queue": 1,
+        "inter_op_threads": 1,
+        "intra_op_threads": 1
+    }
+
+For a walkthrough of how to colocate a Model, navigate to the :ref:`Colocated Orchestrator<link>` for
+instructions.
+
+=====
+Files
+=====
+--------
+Overview
+--------
+The ``Model.attach_generator_files()`` function enables users to write model
+parameters to a file or attachment a file of parameters with values to the
+Model entity for use within the simulation.
+Users can specify whether to copy, symlink, or configure files by passing parameters
+one of the following parameters: `to_copy`, `to_symlink`, or `to_configure`.
+It's worth noting that invoking
+this function, ``attach_generator_files()``, post-attachment overwrites the existing list of entity files.
+
+To write Model parameters to a file and attach to the Model, pass one of the
+following values to the helper function:
+
+* `to_copy` (list, optional): Files that are copied into the path of the entity.
+* `to_symlink` (list, optional): Files that are symlinked into the path of the entity.
+
+To attach a file to the Model and read the contents for use within the simulation, pass the
+following value to the helper function:
+
+* `to_configure` (list, optional): Designed for text-based model input files,
+  "to_configure" is exclusive to models. These files contain parameters for
+  the model, with customizable tags corresponding to values users intend to
+  modify. The default tag is a semicolon (e.g., THERMO = ;10;).
+
+In the `Example` subsection, we provide an example illustrating using the value `to_configure`
```

Comment:
Can we link? If it's immediate (it is), could we say "see example"?

Reply:
Probably follow-up w/ @MattToast for wording, but I think we can improve this to indicate that `to_configure` is used to specify _a template_ for supplying a dynamic set of values...

Reply:
so I actually spoke with Matt E and I got the functionality incorrect, to_configure, stores parameters to a file from the 'params' argument